src/ORM/Behavior/TreeBehavior.php

```diff
@@ -76,6 +77,23 @@ class TreeBehavior extends Behavior
     ];
 
     /**
+     * {@inheritDoc}
+     */
+    public function __construct(Table $table, array $config = [])
+    {
+        parent::__construct($table, $config);
+        $connection = $table->connection();
+        if ($connection !== null) {
+            $driver = $connection->driver();
+            $this->config('leftField', $driver->quoteIdentifier($this->config('left')));
+            $this->config('rightField', $driver->quoteIdentifier($this->config('right')));
+        } else {
+            $this->config('leftField', $this->config('left'));
+            $this->config('rightField', $this->config('right'));
```

Comment:
@markstory, So what do you propose. One possible supution would implement AssignmentExpression (class that will serve such sort of expressions). Possible alternative is have class that join mutiple expressions incuding plain strings into single string. In this case we will have two Identifier expressions with  '=' and '+1' strings.